### PR TITLE
Do and commit

### PR DESCRIPTION
**Pull request by munterkalmsteiner**
Replace call to commit() with doAndCommit().

### CODE REVIEW

File: bundles/specmate-auth/src/com/specmate/auth/internal/PersistentSessionService.java [hunk 1]
```diff
@@ -111,10 +123,15 @@ public AccessRights getTargetAccessRights(String token) throws SpecmateException
...
 	@Override
-	public void delete(String token) throws SpecmateException {
-		UserSession session = (UserSession) sessionTransaction.getObjectById(getSessionID(token));
-		SpecmateEcoreUtil.detach(session);
-		sessionTransaction.commit();
+	public void delete(String token) throws SpecmateException, SpecmateValidationException {
+		sessionTransaction.doAndCommit(new IChange<Object>() {
+			@Override
+			public Object doChange() throws SpecmateException, SpecmateValidationException {
+				UserSession session = (UserSession) sessionTransaction.getObjectById(getSessionID(token));
```
Comment (junkerm) [inline bundles/specmate-auth/src/com/specmate/auth/internal/PersistentSessionService.java]: you could pull this line out, also
BTW: What would happen if the the object with the id is not found?

File: bundles/specmate-connectors/src/com/specmate/connectors/internal/ConnectorTask.java [hunk 2]
```diff
@@ -63,16 +65,23 @@ private void syncRequirementsFromSources() {
 					greatestUnhandledIndex = upperIndexExclusive;
 					List<Requirement> tosync = Arrays.asList(current);
 					syncContainers(localContainer, tosync, source);
-					transaction.commit();
+
+					transaction.doAndCommit(new IChange<Object>() {
+						@Override
+						public Object doChange() throws SpecmateException, SpecmateValidationException {
+							return null;
```
Comment (junkerm) [inline bundles/specmate-connectors/src/com/specmate/connectors/internal/ConnectorTask.java]: that seems strange, there is nothing happening inside the callback method. Probably the call to syncContainers should go there

File: bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyDataService.java [hunk 3]
```diff
@@ -87,7 +88,12 @@ private void fillDummyData() throws SpecmateException {
 			transaction.getResource().getContents().add(testFolder);
 
 			try {
-				transaction.commit();
+				transaction.doAndCommit(new IChange<Object>() {
+					@Override
+					public Object doChange() throws SpecmateException {
+						return null;
```
Comment (junkerm) [inline bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyDataService.java]: same here, the transaction.getResource.... line should go inside the method call

File: bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyDataService.java [hunk 4]
```diff
@@ -109,7 +115,12 @@ private void fillDummyData() throws SpecmateException {
...
```
Comment (junkerm) [inline bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyDataService.java]: see above

File: bundles/specmate-integration-test/src/com/specmate/test/integration/CDOPersistencyShutdownTest.java [hunk 5]
```diff
@@ -100,20 +107,26 @@ private Folder getTestFolder() {
...
 		try {
 			folder.setId(Long.toString(System.currentTimeMillis()));
 		} catch (Exception e) {
 			throw new SpecmateException("Could not access transaction", e);
 		}
-		transaction.commit();
+		transaction.doAndCommit(new IChange<Object>() {
+			@Override
+			public Object doChange() throws SpecmateException, SpecmateValidationException {
+				return null;
```
Comment (junkerm) [inline bundles/specmate-integration-test/src/com/specmate/test/integration/CDOPersistencyShutdownTest.java]: see above

File: bundles/specmate-integration-test/src/com/specmate/test/integration/IntegrationTestBase.java [hunk 6]
```diff
@@ -70,10 +72,15 @@ protected void configureDBProvider(Dictionary<String, Object> properties) throws
...
 
-	private void clearPersistency() throws SpecmateException {
+	private void clearPersistency() throws SpecmateException, SpecmateValidationException {
 		ITransaction transaction = persistency.openTransaction();
 		transaction.getResource().getContents().clear();
-		transaction.commit();
+		transaction.doAndCommit(new IChange<Object>() {
+			@Override
+			public Object doChange() throws SpecmateException, SpecmateValidationException {
+				return null;
```
Comment (junkerm) [inline bundles/specmate-integration-test/src/com/specmate/test/integration/IntegrationTestBase.java]: see above

File: bundles/specmate-integration-test/src/com/specmate/test/integration/ProjectConfigServiceTest.java [hunk 7]
```diff
@@ -103,15 +105,20 @@ private void testAllNewLibraryFolders_initData() throws SpecmateException {
 			testB.setName(projectNames[1]);
 			root.add(testB);
 
-			trans.commit();
+			trans.doAndCommit(new IChange<Object>() {
+				@Override
+				public Object doChange() throws SpecmateException, SpecmateValidationException {
+					return null;
```
Comment (junkerm) [inline bundles/specmate-integration-test/src/com/specmate/test/integration/ProjectConfigServiceTest.java]: see above

File: bundles/specmate-connectors/src/com/specmate/connectors/internal/ConnectorTask.java (hunk 2 above)
Comment (munterkalmsteiner) [reply bundles/specmate-connectors/src/com/specmate/connectors/internal/ConnectorTask.java]: Fixed, as well as all other instances.

**Comment by munterkalmsteiner** [comment]
@junkerm Fixed your comments. If you agree, it is ready to merge.

**Pull request merged**